pkg/hds/authn/callbacks.go

```diff
@@ -20,18 +22,28 @@ const authorization = "authorization"
 
 // Inspired by pkg/xds/auth/callbacks.go
 
-func NewCallbacks(resManager core_manager.ResourceManager, authenticator xds_auth.Authenticator) hds_callbacks.Callbacks {
+type DPNotFoundRetry struct {
+	Backoff  time.Duration
+	MaxTimes uint
```

Comment:
Is this the maximum number of retries? What happens next, should the caller try again?

Reply:
Yes, it's the same behavior as it is right now, but "smoothen out" meaning that we are making some of the retries instead of the client.